python/pyspark/pandas/tests/test_stats.py

```diff
@@ -180,6 +176,10 @@ def test_axis_on_dataframe(self):
                 },
                 index=range(10, 15001, 10),
             )
+            # There is a regression in Pandas 2.1.0,
+            # so we should manually cast to float until the regression is fixed.
+            # See https://github.com/pandas-dev/pandas/issues/55194.
+            pdf = pdf.astype(float)
```

Comment:
If this is a `Pandas` bug, it seems that we need another TODO Spark JIRA ID to clean up this after Pandas community fixes it.

Reply:
Sounds good! Just created https://issues.apache.org/jira/browse/SPARK-45228 for tracking, and added comments to the test.